### ISSUE TITLE
Add option to open DevTools

### ISSUE THREAD

**Issue by parro-it** [sindresorhus/electron-debug#32]
The more I use electron-debug, the more I'm annoyed by this use case.
I open DevTools with toggle shortcut, then often it got covered by other windows, but still open.
After a while, I often forgot it is open and I press toggle key to show DevTools, but since it's already opened, it close.

I think it should be useful to add an option to just open it if closed, or otherwise focus it.
I don't think there is a specific method in electron to focus, we could ask for it (or maybe it could suffice to re-open it, I have to try).

**Comment by sindresorhus** [comment]
Can't say that's been an issue for me, but I'm open to adding something for this. The downside of adding it as an option is that it would permanently remove ability to close dev tools from the keyboard. Wouldn't it be better to just add an alternative keyboard shortcut?

> I don't think there is a specific method in electron to focus

I think you can focus this http://electron.atom.io/docs/v0.36.8/api/web-contents/#webcontentsdevtoolswebcontents

**Comment by parro-it** [comment]
> Wouldn't it be better to just add an alternative keyboard shortcut?

Yes, I mean to add a new shortcut, keeping all existing stuff, that is still useful

**Comment by parro-it** [comment]
> I think you can focus this http://electron.atom.io/docs/v0.36.8/api/web-contents/#webcontentsdevtoolswebcontents

Already tried, but [BrowserWindow.fromWebContents](https://github.com/atom/electron/blob/master/docs/api/browser-window.md#browserwindowfromwebcontentswebcontents) always return null for a DevTools window, so no luck there.

**Comment by sindresorhus** [comment]
> Already tried, but BrowserWindow.fromWebContents always return null for a DevTools window, so no luck there.

Ok. Can you open an Electron issue and cross-link it here?

**Comment by sindresorhus** [comment]
> Yes, I mean to add a new shortcut, keeping all existing stuff, that is still useful

:+1:

**Comment by parro-it** [comment]
> Ok. Can you open an Electron issue and cross-link it here?

Sure!

**Comment by parro-it** [comment]
I opened issue upstream on electron: https://github.com/atom/electron/issues/4625

**Comment by parro-it** [comment]
My issue on  Electron appear to be blocked by [this other](https://github.com/atom/electron/issues/2835).

I done some experiments [here](https://github.com/parro-it/debug-menu/blob/open-devtools-native/index.js) but they are unfortunate...

I published a version of the method directly  on [debug-menu](https://github.com/parro-it/debug-menu/blob/master/index.js#L50), I  will refactor that when upstream issue is solved and them move to `electron-debug`,

**Comment by vinnymac** [comment]
If I read this issue right, all you want to do is open it when it is closed, focus it when it is open.

I do this in my app already like this.

```javascript
const firstWindow = BrowserWindow.getAllWindows()[0]
if (!firstWindow) return

const isDevToolsOpened = firstWindow.isDevToolsOpened()
const isDevToolsFocused = firstWindow.isDevToolsFocused()

if (isDevToolsOpened && isDevToolsFocused) {
  firstWindow.closeDevTools()
} else if (isDevToolsOpened && !isDevToolsFocused) {
  firstWindow.devToolsWebContents.focus()
} else {
  firstWindow.openDevTools()
}
```

You could remove the first test if you don't want it closing on you. I use this with the `Shift+Ctrl/Cmd+I` shortcuts and it hasn't given me any trouble.

**Comment by sindresorhus** [comment]
Alright, let's do this by default when the DevTools is undocked. Apparently, that's how the native Chrome DevTools work too.

**Comment by Nantris** [comment]
This appears to be completed as the project supports `showDevTools`.

**Comment by sindresorhus** [comment]
Opening DevTools on each window is already covered by the `showDevTools` option.

The other suggestion in this thread was to focus the DevTools window instead of closing it when it is already open. That cannot ship on its own. The shortcut cannot be detected while DevTools has focus (#76), so focusing instead of closing would leave no keyboard way to close DevTools at all, which is worse than today.

The readme now documents the focus limitation. Closing as not planned.
